src/vm/runtimehandles.cpp

```diff
@@ -340,7 +340,7 @@ FCIMPL3(void, RuntimeMethodHandle::CheckLinktimeDemands, ReflectMethodObject *pM
 
         if (isDecoratedTargetSecurityTransparent && isAttributeSecurityCritical)
         {
-            if (!DoAttributeTransparencyChecks(pCallee->GetAssembly(), pDecoratedModule->GetAssembly()))
+            if (Security::IsTransparencyEnforcementEnabled() && !DoAttributeTransparencyChecks(pCallee->GetAssembly(), pDecoratedModule->GetAssembly()))
```

Comment:
I would be better (for perf) to this as the very first thing at the start of the method (even before QCALL or HMF frame).